internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -163,8 +161,11 @@ func (hr *HypershiftAgentServiceConfigReconciler) Reconcile(origCtx context.Cont
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	// Create a map for properties that are shared between ASC objects
+	ascProperties := make(map[string]interface{})
```

Comment:
Is moving this out solving a bug of some sort? This really feels like it deserves its own PR and an issue, or at least a separate commit.